src/components/CheckIn/CheckIn.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react'
+import { uuid4 } from '@sentry/utils'
+import { ConfirmDialog } from './internal/ConfirmDialog'
+import { DummyCheckInButton } from './internal/DummyCheckInButton/DummyCheckInButton'
+import { Debug } from './internal/Debug/Debug'
+import { DummyCamera } from './internal/DummyCamera'
+import { Typography } from '@material-ui/core'
+
+type CheckInType = 'event' | 'session'
+
+type Props = {
+  checkInType: CheckInType
+}
+
+export const CheckIn: React.FC<Props> = ({ checkInType }) => {
+  const [storedKeys, setStoredKeys] = React.useState<string[]>([])
+  const [open, setOpen] = useState(false)
+  useEffect(() => {
+    for (const key in localStorage) {
+      if (key.startsWith('check_in_')) {
+        setStoredKeys((prev) => (prev.includes(key) ? prev : [...prev, key]))
+      }
+    }
+
```

Comment:
ここは、backend送信前にreloadしたりなどしてlocalStorageに残っているものを救済するためのコードだと思いますが、
componentの初期化処理にするのが良いと思います。
- storedKeysの更新をtriggerとして実行されるside effectの中でsetStoredKeysを呼ぶのは極力避けた方が良いです
- 初回以外はstoredKeysとlocalStorageは同期しているはずなので、初回だけで良いはずです。であれば、初回だけ呼ばれるように実装した方が意図が明確になります。
- あと、setStoredKeysをcallするのも、都度呼ぶよりもまとめて呼んだほうが良いです。

```suggestion
  useEffect(() => {
    const keys = Object.keys(localStorage).filter((key) =>
      key.startsWith('check_in_'),
    )
    setStoredKeys(keys)
  }, [])

  useEffect(() => {
```